sympy/solvers/ode.py

```diff
@@ -346,6 +346,160 @@
     )
 
 
+class Solver:
+    def __init__(self, eq, func, hint="default", simplify=True, ics=None):
+        # Store the inputs passed to dsolve
+        self.eq, self.func = self.pre_process(eq, func)
+        self.x = self.func.args[0]
+        self.order = ode_order(eq, self.func.func(self.x))
+        self.ics = ics
+        self.simplify = simplify
+        self.hint = hint
+
+    def pre_process(self, eq, func):
+        # To convert equality to expression and find function f if not given by user.
+        # Same as that used by _desolve()
+        if isinstance(eq, Equality):
```

Comment:
This should not be necessary as this class will be used by `dsolve` which will have already done this preprocessing.